src/Avalonia.Base/Reactive/AvaloniaPropertyObservable.cs

```diff
@@ -44,8 +44,13 @@ private void PropertyChanged(object sender, AvaloniaPropertyChangedEventArgs e)
         {
             if (e.Property == _property)
             {
-                _value = (T)e.NewValue;
-                PublishNext(_value);
+                var newValue = e.Sender.GetValue(e.Property);
```

Comment:
I got curious about this code since it will box value types multiple times. I've added a simple check if `e.Property` is `AvaloniaProperty<T>` and in ControlCatalog 60% of times it is. This allows this code to avoid all boxing and just use typed values. We still need a codepath for `AvaloniaPropertyObservable<object>` which is used by bindings code (hopefully to be typed as well eventually).